### PR TITLE
Auto: implement issue #577

### LINKED ISSUE #577: Add AiDraft Pydantic schemas and AiDraftService
Subtask of #50

Define request/response Pydantic models in src/models/ai_draft.py: DraftRequest(type: Literal[email,sms], subject?: str, tone: Literal[professional,friendly,urgent], context: DraftContext with customer_id/opportunity_id/TemplateType), DraftResponse(body, suggested_actions). Create AiDraftService in src/services/ai_draft_service.py that calls the AI agent framework (from #41) to generate drafts. Return the generated draft and suggested actions from the service. No router yet.

Dev-Plan: docs/dev-plan/50-automation/0577-add-aidraft-pydantic-schemas-and-aidraftservice.md

### PR DESCRIPTION
**Pull request by yihua-zhuo**
Closes #577

Plan: `.plans/issue-577.md`

This PR was auto-implemented by Claude Code in three stages:

1. **Plan** — Claude wrote a detailed implementation plan to `.plans/issue-577.md`.
2. **Implement** — Claude executed the plan, ran `ruff check src/` + the unit and integration test suites, and iterated until all three were green.
3. **Self-review** — Claude reviewed its own diff against `codereview.md` (the project review checklist) and fixed any rule violations found, re-running the test suite if any changes were made.

PR-time CI (`Dev Agent CI`) re-verifies independently before this PR is mergeable.

Please review before merging.

## Release Notes

* **New Features**
  * Added AI-powered draft generation for emails and SMS with customizable tone options (professional, friendly, urgent).
  * Drafts now include suggested actions for improved user engagement.
  * Enhanced validation to ensure required fields are present (e.g., email subjects).

[Review Change Stack](https://app.coderabbit.ai/change-stack/yihua-zhuo/agent-job/pull/695?utm_source=github_walkthrough&utm_medium=github&utm_campaign=change_stack)

### CODE REVIEW

File: tests/integration/test_ai_draft_service_integration.py [hunk 1]
```diff
@@ -0,0 +1,197 @@
...
+        await _seed_opportunity(async_session, tenant_id, customer_id=cust.id, name="Big Deal")
+
+        svc = AiDraftService(async_session, gateway=StubAIChatGateway())
+        request = DraftRequest(
+            type=DraftType.EMAIL,
+            subject="Deal update",
+            tone=ToneType.PROFESSIONAL,
+            context=DraftContext(
+                customer_id=cust.id,
+                opportunity_id=cust.id,  # use cust.id as proxy (opportunity created above)
```
Comment (yihua-zhuo) [inline tests/integration/test_ai_draft_service_integration.py]: **CRITICAL** (critical)

**Issue:** Test test_generate_draft_with_opportunity_id sets opportunity_id=cust.id instead of opportunity_id=opp.id, so the opportunity entity is never actually referenced despite being seeded; the test passes only because cust.id == opp.customer_id, masking a potential service bug (Rule 64)

**Suggestion:** Change opportunity_id=cust.id to opportunity_id=opp.id, and verify opp was created with await _seed_opportunity(...). If opp.id is needed, capture it: opp = await _seed_opportunity(...); then request.context.opportunity_id = opp.id

Comment (yihua-zhuo) [inline tests/integration/test_ai_draft_service_integration.py]: **CRITICAL** (critical)

**Issue:** Integration test with opportunity_id does not assert that the AI prompt includes opportunity context or that the opportunity data (e.g. stage, amount, name) influences the draft, so a missing opportunity_id in _build_prompt would go undetected (Rule 64)

**Suggestion:** Add assertions verifying the draft body or suggested_actions reflect opportunity data (e.g. 'Big Deal' name appears in the prompt or actions)

File: tests/unit/test_ai_draft_service.py [hunk 2]
```diff
@@ -0,0 +1,250 @@
...
+from services.ai_draft_service import AiDraftService
+
+
+# ---------------------------------------------------------------------------
+# Mock session (no DB needed — customer lookup is the only query)
+# ---------------------------------------------------------------------------
+
+
+@pytest.fixture
+def mock_db_session():
```
Comment (yihua-zhuo) [inline tests/unit/test_ai_draft_service.py:62]: **CRITICAL** (critical)

**Issue:** Unit test defines a file-local mock_db_session with raw MagicMock/AsyncMock instead of the composable MockState + domain handler pattern from tests/unit/conftest.py (Rule 135)

**Suggestion:** Use make_mock_session([make_customer_handler(state)]) from tests.unit.conftest or tests.unit.domain_handlers.customers, with MockState() as state

File: tests/unit/test_ai_draft_service.py [hunk 3]
```diff
@@ -0,0 +1,250 @@
...
+    return session
+
+
+# ---------------------------------------------------------------------------
+# Mock gateway
+# ---------------------------------------------------------------------------
+
+
+@pytest.fixture
+def mock_gateway():
```
Comment (yihua-zhuo) [inline tests/unit/test_ai_draft_service.py]: **CRITICAL** (critical)

**Issue:** Unit test defines a file-local mock_gateway with raw MagicMock/AsyncMock instead of using a composable stub class pattern (Rule 135)

**Suggestion:** Replace mock_gateway fixture with a stub class (e.g. StubAIChatGateway) matching the pattern in the integration test file

File: tests/unit/test_ai_draft_service.py [hunk 4]
```diff
@@ -0,0 +1,250 @@
...
+
+        mock_gateway.chat.assert_called_once()
+        call_args = mock_gateway.chat.call_args
+        messages = call_args[0][0]
+        assert isinstance(messages, list)
+        assert len(messages) > 0
+        assert any(m["role"] == "system" for m in messages)
+        assert any(m["role"] == "user" for m in messages)
+
+    async def test_maps_actions_from_ai_response(self, ai_draft_service, mock_db_session, mock_gateway, draft_request):
```
Comment (yihua-zhuo) [inline tests/unit/test_ai_draft_service.py]: **CRITICAL** (critical)

**Issue:** Unit tests that set mock_db_session.execute.return_value = mock_result never validate that the SQL query bind parameters (tenant_id, customer_id) match what AiDraftService.generate_draft actually passes, enabling silently incorrect tests (Rule 133)

**Suggestion:** Add assert statements in handlers to verify params['tenant_id'] and params['id'] are inspected, or use the domain handler pattern which validates bind keys against real SQL text

Comment (yihua-zhuo) [inline tests/unit/test_ai_draft_service.py]: **CRITICAL** (critical)

**Issue:** Unit tests mock a customer lookup without asserting that tenant_id filtering is enforced, leaving cross-tenant isolation completely untested (Rule 134)

**Suggestion:** Add a cross-tenant negative test: use the same mock to assert that tenant_id=999 returns None for a customer whose id=1, confirming the service correctly gates tenant data

File: .plans/issue-577.md [hunk 5]
```diff
@@ -0,0 +1,58 @@
...
+   - Test `generate_draft` raises `ValidationException` when gateway returns empty reply
+   - Mock `AIChatGateway` in tests that need it via `AiDraftService(mock_db_session, gateway=mock_gateway)`
+   - Use `MockState`, `MockRow`, `MockResult` as needed from `conftest.py`
+
+5. **Create integration test `tests/integration/test_ai_draft_service_integration.py`**:
+   - Fixtures: `db_schema`, `tenant_id`, `async_session`
+   - Use `_seed_customer` / `_seed_opportunity` helpers to populate context data before calling the service
+   - Test `generate_draft` with a real session and a real (or test-double) gateway
+   - Test error case: call `generate_draft` with a non-existent `customer_id` — `ValidationException` or `NotFoundException` depending on where validation occurs
+   - Test `generate_draft` with `opportunity_id` provided and `opportunity_id` omitted
```
Comment (yihua-zhuo) [inline .plans/issue-577.md:44]: **WARNING** (suggestion)

**Issue:** Plan lists 'Test `generate_draft` with `opportunity_id` provided and `opportunity_id` omitted' but the integration test `test_generate_draft_with_opportunity_id` passes `opportunity_id=cust.id` (customer ID) instead of the actual seeded opportunity's ID, and never validates the opportunity context was used.

**Suggestion:** In `tests/integration/test_ai_draft_service_integration.py` line 157: capture the returned opportunity from `_seed_opportunity(...)` and pass `opportunity_id=opp.id` instead of `opportunity_id=cust.id`.

Comment (yihua-zhuo) [inline .plans/issue-577.md:44]: **WARNING** (suggestion)

**Issue:** Plan does not include a cross-tenant isolation test for `generate_draft`, which is required by Rule 126 for any new multi-tenant data path.

**Suggestion:** Add an integration test that seeds a customer under `tenant_id=A`, then calls `generate_draft(request, tenant_id=B)` and asserts `NotFoundException` is raised.

File: src/models/ai_draft.py [hunk 6]
```diff
@@ -0,0 +1,56 @@
...
+    label: str
+    action_type: str
+    payload: dict
+
+
+class DraftResponse(BaseModel):
+    body: str
+    suggested_actions: list[SuggestedAction]
+
+    def to_dict(self) -> dict:
```
Comment (yihua-zhuo) [inline src/models/ai_draft.py]: **INFO** (suggestion)

**Issue:** Plan step 23 says to follow the same pattern as `ChatResponse.to_dict` in `ai.py`, which declares `-> dict[str, Any]`; `DraftResponse.to_dict` declares bare `-> dict` with no generic parameters.

**Suggestion:** Change `def to_dict(self) -> dict:` to `def to_dict(self) -> dict[str, Any]:` to match the `ai.py` pattern the plan references.

File: src/services/ai_draft_service.py [hunk 7]
```diff
@@ -0,0 +1,71 @@
+"""AI Draft service — generates email/SMS drafts via AIChatGateway."""
+
+from sqlalchemy import select
```
Comment (yihua-zhuo) [inline src/services/ai_draft_service.py:3]: **INFO** (suggestion)

**Issue:** `select` is imported but never referenced; `CustomerModel` queries use `where()` directly without an explicit `select()` call wrapper.

**Suggestion:** Remove `select` from the `from sqlalchemy import` line, or use `from sqlalchemy import select` if it will be used in future edits.

File: src/services/ai_draft_service.py [hunk 8]
```diff
@@ -0,0 +1,71 @@
...
+                label=action.get("label", ""),
+                action_type=action.get("type", ""),
+                payload=action,
+            )
+            for action in (response.actions or [])
+        ]
+
+        return DraftResponse(body=response.reply, suggested_actions=suggested_actions)
+
+    async def _build_prompt(self, request: DraftRequest, tenant_id: int) -> list[dict[str, str]]:
```
Comment (yihua-zhuo) [inline src/services/ai_draft_service.py:46]: **WARNING** (suggestion)

**Issue:** No cross-tenant isolation test exists for `generate_draft`. The method correctly filters by `tenant_id` at line 23, but Rule 126 requires negative tests proving tenant A cannot access tenant B's customer.

**Suggestion:** Add an integration test: seed a customer under tenant X, call `generate_draft(..., tenant_id=Y)` with Y != X, and assert `NotFoundException`.

File: tests/integration/test_ai_draft_service_integration.py [hunk 9]
```diff
@@ -0,0 +1,197 @@
...
+Run against a real PostgreSQL database (DATABASE_URL env var):
+    DATABASE_URL="postgresql+asyncpg://..." pytest tests/integration/test_ai_draft_service_integration.py -v
+
+Requires DATABASE_URL (or TEST_DATABASE_URL) pointing at a live Postgres instance.
+Each test gets a fresh schema via TRUNCATE CASCADE (see conftest.py).
+"""
+
+from __future__ import annotations
+
+from unittest.mock import AsyncMock, MagicMock
```
Comment (yihua-zhuo) [inline tests/integration/test_ai_draft_service_integration.py]: **WARNING** (suggestion)

**Issue:** AsyncMock and MagicMock are imported on line 13 but never used; the integration test only uses stub gateway classes, not unittest.mock (Rule 17)

**Suggestion:** Remove the unused 'from unittest.mock import AsyncMock, MagicMock' import line

Comment (yihua-zhuo) [inline tests/integration/test_ai_draft_service_integration.py]: **WARNING** (suggestion)

**Issue:** Integration test imports _call_gateway directly from StubAIChatGateway and uses it in tests, but never asserts on gateway method calls or inspects the messages sent to it, limiting coverage of prompt construction (Rule 64)

**Suggestion:** Add assertions on the messages passed to gateway or verify the generated prompt structure includes subject, tone, customer_id, template_type, and opportunity_id when applicable

File: tests/unit/test_ai_draft_service.py [hunk 10]
```diff
@@ -0,0 +1,250 @@
...
+
+        result = await ai_draft_service.generate_draft(draft_request, tenant_id=1)
+
+        assert isinstance(result, DraftResponse)
+        assert result.body == "This is a draft email body."
+        assert len(result.suggested_actions) == 2
+        assert result.suggested_actions[0].label == "View Customer"
+        assert result.suggested_actions[0].action_type == "navigate"
+
+    async def test_calls_gateway_chat_with_messages(
```
Comment (yihua-zhuo) [inline tests/unit/test_ai_draft_service.py]: **WARNING** (suggestion)

**Issue:** test_calls_gateway_chat_with_messages only asserts that 'system' and 'user' messages exist; it does not verify the prompt content (subject line, tone, customer_id, template_type, opportunity_id) is correctly included, so a broken _build_prompt would go undetected (Rule 64)

**Suggestion:** Add assertions on the user message content, e.g. assert any('Follow-up on our conversation' in m['content'] or 'Subject:' in m['content'] for m in messages)

File: tests/unit/test_ai_draft_service.py [hunk 11]
```diff
@@ -0,0 +1,250 @@
...
+        mock_result.scalar_one_or_none.return_value = mock_customer
+        mock_db_session.execute.return_value = mock_result
+
+        result = await ai_draft_service.generate_draft(draft_request, tenant_id=1)
+
+        assert len(result.suggested_actions) == 2
+        action = result.suggested_actions[1]
+        assert action.label == "Send Draft"
+        assert action.action_type == "send_email"
+        assert action.payload["payload"] == {"draft_id": 1}
```
Comment (yihua-zhuo) [inline tests/unit/test_ai_draft_service.py]: **WARNING** (suggestion)

**Issue:** Test assertion 'action.payload["payload"] == {"draft_id": 1}' is incorrect — the service copies the raw action dict into payload, so payload["payload"] accesses a nested 'payload' key that does not exist; the assertion should be action.payload == {"draft_id": 1} (Rule 15)

**Suggestion:** Change assert action.payload['payload'] == ... to assert action.payload == {"draft_id": 1}

File: tests/unit/test_ai_draft_service.py [hunk 12]
```diff
@@ -0,0 +1,250 @@
...
+    async def test_raises_not_found_for_missing_customer(self, mock_db_session, mock_gateway, draft_request):
+        mock_result = MagicMock()
+        mock_result.scalar_one_or_none.return_value = None
+        mock_db_session.execute.return_value = mock_result
+        svc = AiDraftService(mock_db_session, gateway=mock_gateway)
+
+        with pytest.raises(NotFoundException):
+            await svc.generate_draft(draft_request, tenant_id=1)
+
+    async def test_raises_validation_exception_when_gateway_returns_empty_reply(
```
Comment (yihua-zhuo) [inline tests/unit/test_ai_draft_service.py:224]: **INFO** (suggestion)

**Issue:** Test class TestGenerateDraftErrors uses ai_draft_service fixture which already includes mock_gateway, but constructs a new AiDraftService with a different gateway in two tests (lines 183, 198), causing fixture waste; tests that need a special gateway should pass it as a parameter or use a different fixture composition (Rule 66)

**Suggestion:** Remove ai_draft_service from affected test signatures or refactor to a single parameterized fixture for gateway variants

File: tests/unit/test_ai_draft_service.py [hunk 13]
```diff
@@ -0,0 +1,250 @@
+"""Unit tests for src/services/ai_draft_service.py — AiDraftService business logic."""
+
+from unittest.mock import AsyncMock, MagicMock
```
Comment (yihua-zhuo) [inline tests/unit/test_ai_draft_service.py:3]: **INFO** (suggestion)

**Issue:** AsyncMock and MagicMock are imported but the mock_gateway fixture creates an AsyncMock via MagicMock().chat = AsyncMock(...); prefer importing AsyncMock directly for clarity (Rule 20)

**Suggestion:** Remove 'from unittest.mock import AsyncMock' if only used via MagicMock wrapper, or use AsyncMock directly for gateway.chat

**Comment by yihua-zhuo** [comment]
## Claude Code Review (inline)

**Status:** `pass`
**Inline comments posted:** 10

**Summary:** Reviewed 7 file(s) between origin/master and origin/auto/issue-577.

### Issues without a usable file/line
- `.plans/issue-577.md`: Plan is well-structured and complete; all acceptance criteria are correctly specified.
- `src/models/__init__.py`: Clean re-export file; all six symbols from ai_draft.py plus existing symbols are correctly listed in __all__.
- `src/models/ai_draft.py`: Correct Pydantic schemas with proper StrEnum usage, model_validator enforcing subject required for EMAIL type, and a minimal to_dict() delegating to model_dump().

File: tests/integration/test_ai_draft_service_integration.py [hunk 14]
```diff
@@ -0,0 +1,226 @@
...
+    async def _call_gateway(self, messages, context):
+        return AIResponse(reply="", suggestions=None, actions=None)
+
+
+# ---------------------------------------------------------------------------
+# Helpers
+# ---------------------------------------------------------------------------
+
+
+async def _seed_opportunity(async_session, tenant_id: int, customer_id: int, **overrides) -> OpportunityModel:
```
Comment (yihua-zhuo) [inline tests/integration/test_ai_draft_service_integration.py:61]: **WARNING** (suggestion)

**Issue:** _seed_opportunity is a domain-owned seed helper defined in the integration test file rather than a domain fixtures module, violating rule 125 (feature domains own their registration points).

**Suggestion:** Move _seed_opportunity to tests/integration/domain_fixtures/opportunities.py (or similar domain-owned fixture module) and import it from there.

File: tests/integration/test_ai_draft_service_integration.py [hunk 15]
```diff
@@ -0,0 +1,226 @@
...
+            tone=ToneType.PROFESSIONAL,
+            context=DraftContext(
+                customer_id=cust.id,
+                opportunity_id=opp.id,
+                template_type=TemplateType.EMAIL,
+            ),
+        )
+        result = await svc.generate_draft(request, tenant_id=tenant_id)
+        assert len(result.body) > 0
+        # Verify opportunity name is reflected in the reply (gateway echoes it when opportunity is included)
```
Comment (yihua-zhuo) [inline tests/integration/test_ai_draft_service_integration.py:168]: **INFO** (suggestion)

**Issue:** test_generate_draft_with_opportunity_id asserts 'Big Deal' in result.body, which depends on the StubAIChatGateway echoing the opportunity name into its reply — this is a property of the stub, not the service. The service only passes opportunity_id in the prompt as text; it does not pass the opportunity name.

**Suggestion:** Either remove the assertion on 'Big Deal' in result.body, or restructure the test to verify the opportunity_id appears in the prompt via SpyAIChatGateway rather than relying on the stub's reply content.

File: tests/integration/test_ai_draft_service_integration.py [hunk 16]
```diff
@@ -0,0 +1,226 @@
...
+            tenant_id=tenant_id,
+            name="Cross Tenant Customer",
+            email="crosstenant@example.com",
+            status="active",
+        )
+        async_session.add(cust)
+        await async_session.flush()
+        customer_id_under_tenant_a = cust.id
+
+        # Calling with a different tenant_id_2 should raise NotFoundException
```
Comment (yihua-zhuo) [inline tests/integration/test_ai_draft_service_integration.py:217]: **INFO** (suggestion)

**Issue:** Inline comment '# Calling with a different tenant_id_2 should raise NotFoundException' is misleading — NotFoundException is raised because the customer_id (created under tenant_id) does not exist under tenant_id_2, not because tenant_id_2 is 'different' in a general sense.

**Suggestion:** Rewrite comment to: '# Attempting to access tenant A's customer with tenant B's credentials should raise NotFoundException'

File: tests/unit/domain_handlers/customers.py [hunk 17]
```diff
@@ -67,42 +67,37 @@ def handler(sql_text, params):
...
             return MockResult([MockRow({"id": params.get("id", 1)})])
 
-        if "select" in sql_text and "count" in sql_text and "from customers" in sql_text:
+        if "select" in sql_text and "count(" in sql_text and "from customers" in sql_text:
             tenant_id = params.get("tenant_id")
             count_val = 3 if tenant_id == 1 else 7
             return MockResult([[count_val]])
 
-        if "select" in sql_text and "from customers" in sql_text and "tags" in sql_text and "where id" in sql_text:
+        if "select" in sql_text and "from customers" in sql_text and "tags" in sql_text:
```
Comment (yihua-zhuo) [inline tests/unit/domain_handlers/customers.py:75]: **WARNING** (suggestion)

**Issue:** The tag-filter branch splits on lowercase 'where' to isolate the WHERE clause, but sql_text has already been lowercased via str(sql).lower(), so the split is fine — however, if a future caller passes already-lowercased SQL directly, 'where' may not be found if the original had 'WHERE'. This is fragile.

**Suggestion:** Use sql_text.split('where', 1)[1] only — already correct. Additionally guard the whole block with 'where' in sql_text (already done) so None is returned when no WHERE clause exists at all.

File: tests/unit/domain_handlers/customers.py [hunk 18]
```diff
@@ -67,42 +67,37 @@ def handler(sql_text, params):
...
+            tenant_id_param = params.get("tenant_id") or params.get("tenant_id_1")
+            if customer_id is not None:
+                # Only return state-seeded customers — no fallback fixtures
+                if customer_id in state.customers:
+                    rec = state.customers[customer_id]
+                    if rec.get("tenant_id") == tenant_id_param:
+                        return MockResult([MockRow(rec.copy())])
+                return MockResult([])
+
+        if "select" in sql_text and "from customers" in sql_text and "id" not in params and "id_1" not in params:
```
Comment (yihua-zhuo) [inline tests/unit/domain_handlers/customers.py:100]: **INFO** (suggestion)

**Issue:** The fallback fixtures (Customer A/B) are always appended to tenant_filter==2's list, then cleared. This means for tenant_id 2 the result is always empty rows from seeded state (or empty overall), which correctly handles the zero-state case for that tenant — not a bug but non-obvious.

**Suggestion:** Add a brief comment explaining why tenant_filter==2 is cleared: '# tenant 2 has no seeded fallback in this handler'.

File: tests/unit/domain_handlers/customers.py [hunk 19]
```diff
@@ -67,42 +67,37 @@ def handler(sql_text, params):
...
-                    "updated_at": None,
-                },
-            }
-            if customer_id in fixtures:
-                return MockResult([MockRow(fixtures[customer_id].copy())])
-            return MockResult([])
-
-        if "select" in sql_text and "from customers" in sql_text and "where id" not in sql_text:
+        # Single-customer lookup by id (recognized by 'where customers.id' in SQL text)
+        if "from customers" in sql_text and "where" in sql_text and "customers.id" in sql_text:
```
Comment (yihua-zhuo) [inline tests/unit/domain_handlers/customers.py:89]: **INFO** (suggestion)

**Issue:** The single-customer lookup at line 89 matches on 'customers.id' in sql_text but the subsequent tenant check at line 96 uses loose equality (== with or without None), which is correct but relies on implicit None-coalescing. The logic works because params.get() returns None for missing keys.

**Suggestion:** No change needed — behavior is correct. Consider adding a type hint to handler() for clarity.

File: tests/unit/test_ai_draft_service.py [hunk 20]
```diff
@@ -0,0 +1,325 @@
...
+    async def chat(self, messages):
+        return AIResponse(reply="", suggestions=None, actions=None)
+
+
+class SpyAIChatGateway:
+    """Spy gateway that records messages and delegates to StubAIChatGateway."""
+
+    def __init__(self):
+        self.last_messages = None
+        self._inner = StubAIChatGateway()
```
Comment (yihua-zhuo) [inline tests/unit/test_ai_draft_service.py:108]: **WARNING** (suggestion)

**Issue:** SpyAIChatGateway calls self._inner.chat(messages) but AiDraftService.generate_draft() calls self.gateway.chat(messages) with no context argument, while the stub's _call_gateway path passes context={} — SpyAIChatGateway tests the chat() wrapper which internally calls _call_gateway, so the test exercises the correct interface, but the spy itself bypasses the actual _call_gateway call path used by the service.

**Suggestion:** Either rename or restructure so the spy directly records the _call_gateway call (matching what the service actually invokes), or confirm the current test intent is only to verify messages structure and document the minor coverage gap.

File: tests/unit/test_ai_draft_service.py [hunk 21]
```diff
@@ -0,0 +1,325 @@
...
+        self, mock_db_session, draft_request
+    ):
+        """Gateway returns whitespace-only reply → ValidationException."""
+        svc = AiDraftService(mock_db_session, gateway=StubAIChatGateway(reply="   \n\t  "))
+        with pytest.raises(ValidationException):
+            await svc.generate_draft(draft_request, tenant_id=1)
+
+    async def test_cross_tenant_isolation(self):
+        """Tenant A cannot access tenant B's customer — NotFoundException raised."""
+        customer_row = MockCustomerRow(customer_id=1, tenant_id=1)
```
Comment (yihua-zhuo) [inline tests/unit/test_ai_draft_service.py:242]: **INFO** (suggestion)

**Issue:** test_cross_tenant_isolation uses MagicMock with a manual execute side_effect scoped to tenant_id==1 vs. other. This is a valid one-off mock for the cross-tenant case but is narrower than the composable handler pattern — acceptable per rule 135 but worth noting.

**Suggestion:** No action required; this inline mock is appropriate for the isolation test scenario where only one specific SQL behavior needs to be verified.

File: tests/unit/test_ai_draft_service.py [hunk 22]
```diff
@@ -0,0 +1,325 @@
...
+from services.ai_draft_service import AiDraftService
+from tests.unit.conftest import MockResult, MockRow
+
+
+# ---------------------------------------------------------------------------
+# Mock session — use handler pattern from conftest + direct execute patching
+# ---------------------------------------------------------------------------
+
+
+class MockCustomerRow(MockRow):
```
Comment (yihua-zhuo) [inline tests/unit/test_ai_draft_service.py:27]: **INFO** (suggestion)

**Issue:** MockCustomerRow defines a complete customer record mapping but does not include fields that may exist on the real CustomerModel (e.g., created_at/updated_at are None here, matching the real model defaults, but any new fields added to CustomerModel will silently not appear in MockCustomerRow).

**Suggestion:** Add a comment at the top of MockCustomerRow noting that fields must be kept in sync with CustomerModel, or derive the field list from a constant shared with the ORM model.
